modules/ppcp-wc-gateway/src/Helper/PayUponInvoiceHelper.php

```diff
@@ -92,12 +91,12 @@ private function is_valid_currency(): bool {
 	}
 
 	/**
-	 * Checks whether PUI is enabled.
+	 * Checks whether PUI gateway is enabled.
 	 *
-	 * @return bool True if PUI is active, otherwise false.
-	 * @throws NotFoundException If problem when checking the settings.
+	 * @return bool True if PUI gateway is enabled, otherwise false.
 	 */
-	public function is_pui_enabled(): bool {
-		return $this->settings->has( 'products_pui_enabled' ) && $this->settings->get( 'products_pui_enabled' );
+	public function is_pui_gateway_enabled(): bool {
+		$gateway_settings = get_option( 'woocommerce_ppcp-pay-upon-invoice-gateway_settings' );
```

Comment:
Better use gateway `enabled` property instead of WP option here, here you have an implementation example: https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-wc-gateway/src/Checkout/DisableGateways.php#L91-L92

Reply:
@Dinamiko I have tried but when you use it too early, like in settings then the `WC()` doesn't exist yet and I am getting the `Fatal error: Uncaught Error: Call to a member function payment_gateways() on null` and I think that's why the same `get_option` is also used [here](https://github.com/woocommerce/woocommerce-paypal-payments/blob/dade07ca95d56680a329edccc62884326ce274fc/modules/ppcp-wc-gateway/src/Helper/PayUponInvoiceHelper.php#L53) and in some other places